README.md

```diff
@@ -1,10 +1,153 @@
+
+```json
+{
+  "item.key.fizzics.1" : {
+    "consume_after_use" : true,
+    "used" : true
+  },
+  "lock.fizzics.1 : {
+    "locked" : true
+  }
+}
+```
+
+The example above shows entries that were written and read by the [Clubhouse](https://github.com/endlessm/clubhouse) and by the [Toolbox](https://github.com/endlessm/hack-toolbox-app). In the example, the Clubhouse has given the player the first key in the Fizzics quest line, which was later used and updated by the Toolbox to unlock the first Fizzics lockscreen. All that interaction between the Clubhouse, the Toolbox and the player's state is done throught the GameStateService.
+
+**NOTE**: the root object of player's state must be a dictionary, while the sub-objects can, in theory, use any kind of serializable objects. When in doubt, make sure to test that your objects are being serialized properly, by manually reading the state file.
+
+**NOTE**: the GameStateService does not provide, nor enforces, any kind content schema or structure. The contents of the state-file are entirely defined by the other components of the desktop. Therefore, it requires an agreement between these components to honour these definitions. For now, it is the price of simplicity.
+
+## The DBus Service
+
+The GameStateService exposes the API consumed by other components of the desktop to access and manipulate the player's state.
+
+### Writing To The State
+
+```javascript
+const {Gio} = imports.gi;
+
+const BusName = 'com.endlessm.GameStateService';
+const BusPath = '/com/endlessm/GameStateService';
+const BusIface = `
+<node>
+  <interface name='com.endlessm.GameStateService'>
+    <method name='Set'>
+      <arg type='s' name='key' direction='in'/>
+      <arg type='v' name='value' direction='in'/>
+    </method>
+  </interface>
+</node>
+`;
+const GameStateProxy = Gio.DBusProxy.makeProxyWrapper(BusIface);
+const gameStateProxy = new GameStateProxy(Gio.DBus.session, BusName, BusPath);
+
+const variant = new GLib.Variant('a{sb}', {locked: false});
+gameStateProxy.SetSync('lock.example.1', variant)
+```
+
+**NOTE**: the killer-feature of this service, is the ability to store and retrieve virtually any kind of serializable data, as these methods take a [GLib.Variant](https://developer.gnome.org/glib/stable/glib-GVariant.html) as the value.
+
+### Reading From The State
+
+```javascript
+const {Gio} = imports.gi;
 
-### Building a Flatpak
+const BusName = 'com.endlessm.GameStateService';
+const BusPath = '/com/endlessm/GameStateService';
+const BusIface = `
+<node>
+  <interface name='com.endlessm.GameStateService'>
+    <method name='Get'>
+      <arg type='s' name='key' direction='in'/>
+      <arg type='v' name='value' direction='out'/>
+    </method>
+  </interface>
+</node>
+`;
+const GameStateProxy = Gio.DBusProxy.makeProxyWrapper(BusIface);
+const gameStateProxy = new GameStateProxy(Gio.DBus.session, BusName, BusPath);
```

Comment:
What about splitting the setup to obtain a `gameStateProxy` object in a "Setup" subsection, to avoid duplicating it in writing state, reading state, monitoring state sections?